tests/test_models.py

```diff
@@ -1,59 +1,120 @@
     assert m.output_shape[1] == 512
 
-    m = load_embedding_model('mel256', 'music', 6144)
+    m = load_audio_embedding_model('mel256', 'music', 6144)
     assert m.output_shape[1] == 6144
 
-    m = load_embedding_model('mel256', 'music', 512)
+    m = load_audio_embedding_model('mel256', 'music', 512)
     assert m.output_shape[1] == 512
 
-    m = load_embedding_model('mel256', 'env', 6144)
+    m = load_audio_embedding_model('mel256', 'env', 6144)
     assert m.output_shape[1] == 6144
 
-    m = load_embedding_model('mel256', 'env', 512)
+    m = load_audio_embedding_model('mel256', 'env', 512)
+    assert m.output_shape[1] == 512
+
+
+def test_get_image_embedding_model_path():
+    embedding_model_path = get_image_embedding_model_path('linear', 'music')
+    assert '/'.join(embedding_model_path.split('/')[-2:]) == 'openl3/openl3_image_linear_music.h5'
+
+    embedding_model_path = get_image_embedding_model_path('linear', 'env')
+    assert '/'.join(embedding_model_path.split('/')[-2:]) == 'openl3/openl3_image_linear_env.h5'
+
+    embedding_model_path = get_image_embedding_model_path('mel128', 'music')
+    assert '/'.join(embedding_model_path.split('/')[-2:]) == 'openl3/openl3_image_mel128_music.h5'
+
+    embedding_model_path = get_image_embedding_model_path('mel128', 'env')
+    assert '/'.join(embedding_model_path.split('/')[-2:]) == 'openl3/openl3_image_mel128_env.h5'
+
+    embedding_model_path = get_image_embedding_model_path('mel256', 'music')
+    assert '/'.join(embedding_model_path.split('/')[-2:]) == 'openl3/openl3_image_mel256_music.h5'
+
+    embedding_model_path = get_image_embedding_model_path('mel256', 'env')
+    assert '/'.join(embedding_model_path.split('/')[-2:]) == 'openl3/openl3_image_mel256_env.h5'
+
+
+def test_load_image_embedding_model():
+    m = load_image_embedding_model('linear', 'music', 8192)
+    assert m.output_shape[1] == 8192
+
+    m = load_image_embedding_model('linear', 'music', 512)
+    assert m.output_shape[1] == 512
+
+    m = load_image_embedding_model('linear', 'env', 8192)
+    assert m.output_shape[1] == 8192
+
+    m = load_image_embedding_model('linear', 'env', 512)
+    assert m.output_shape[1] == 512
+
+    m = load_image_embedding_model('mel128', 'music', 8192)
+    assert m.output_shape[1] == 8192
+
+    m = load_image_embedding_model('mel128', 'music', 512)
+    assert m.output_shape[1] == 512
+
+    m = load_image_embedding_model('mel128', 'env', 8192)
+    assert m.output_shape[1] == 8192
+
+    m = load_image_embedding_model('mel128', 'env', 512)
+    assert m.output_shape[1] == 512
+
+    m = load_image_embedding_model('mel256', 'music', 8192)
+    assert m.output_shape[1] == 8192
+
+    m = load_image_embedding_model('mel256', 'music', 512)
+    assert m.output_shape[1] == 512
+
+    m = load_image_embedding_model('mel256', 'env', 8192)
+    assert m.output_shape[1] == 8192
+
+    m = load_image_embedding_model('mel256', 'env', 512)
```

Comment:
This seems like a relatively weak test? Is there no stronger way to assert whether the model loaded is actually the desired model? e.g. compare architectures or something along these lines?

Reply:
Sure we could compare architectures, and potentially the input types (like making sure there's a Melspectrogram layer with the appropriate number of filters). However, I see no way to test for the content type used to train the model.

Reply:
That's fine, as long as we make a decent effort. Comparing architectures and testing for input types should do the trick.